src/Database.ts

```diff
@@ -412,14 +412,9 @@ export class Database<T = unknown, PAR = unknown> extends TypedEmitter<QmEvents<
      */
     public async all(options?: AllQueryOptions) {
         this.__readyCheck();
-        const everything = await this.model.find({
-            $where: function () {
-                const expiredCheck = !(this.expireAt && this.expireAt.getTime() - Date.now() <= 0);
-                return expiredCheck;
-            }
-        });
+        const everything = await this.model.find({});
         let arb = everything
-            .filter((v) => options?.filter?.({ ID: v.ID, data: v.data }) ?? true)
+            .filter((v) => (options?.filter?.({ ID: v.ID, data: v.data }) ?? true) || !(this.expireAt && this.expireAt.getTime() - Date.now() <= 0))
```

Comment:
This change will affect the result if the filter option is not null
For example, if the filter option is not null, the expiration date will not be taken into consideration if the filter is true because of the `||` operator. Which normally before the change, those expired elements would have been filtered out before.

```suggestion
.filter((v) => !(this.expireAt && this.expireAt.getTime() - Date.now() <= 0))
.filter((v) => options?.filter?.({ ID: v.ID, data: v.data }) ?? true);
```

This is the suggestion I am proposing to result this problem, you can change it but if you decide to keep this or change it make sure to test it to see if it works correctly.

Thanks